File: src/app/modules/m6-StatisticsReports/data-access/models/dashboard-kpi-configuration.dto.ts
import { BaseAuditFields, AggregationMethod, ChartType } from './base.types';

export interface DashboardKpiConfiguration extends BaseAuditFields {
  schoolId: number | null;
  kpiCode: string;
  kpiNameAr: string;
  kpiNameEn: string | null;
  kpiDescription: string | null;
  sourceModule: string;
  sourceTable: string | null;
  sourceField: string | null;
  aggregationMethod: AggregationMethod;
  chartType: ChartType;
  refreshIntervalMinutes: number;
  targetValue: number | null;
  thresholdGreen: number | null;
  thresholdYellow: number | null;
  thresholdRed: number | null;
  alertEnabled: boolean;
  alertRecipientsJson: string | null;
  isActive: boolean;
  displayOrder: number;
  dashboardId: number | null;
}

export type CreateDashboardKpiConfiguration = Omit<DashboardKpiConfiguration, 'id' | 'createdAt' | 'modifiedAt'>;

export type UpdateDashboardKpiConfiguration = Pick<DashboardKpiConfiguration, 'id'> & Partial<Omit<DashboardKpiConfiguration, 'id' | 'createdAt' | 'modifiedAt'>>;

File: 01_Backend/EduMS.Backend/src/EduMS.Application/M6_StatisticsReports/DTOs/DashboardKpiConfigurations/DashboardKpiConfigurationDto.cs
using System;

namespace EduMS.Application.M6_StatisticsReports.DTOs.DashboardKpiConfigurations;

public class DashboardKpiConfigurationDto
{
    public long Id { get; set; }
    public long? SchoolId { get; set; }
    public string KpiCode { get; set; } = string.Empty;
    public string KpiNameAr { get; set; } = string.Empty;
    public string? KpiNameEn { get; set; }
    public string? KpiDescription { get; set; }
    public string SourceModule { get; set; } = string.Empty;
    public string? SourceTable { get; set; }
    public string? SourceField { get; set; }
    public int AggregationMethod { get; set; }
    public int ChartType { get; set; }
    public int RefreshIntervalMinutes { get; set; } = 60;
    public decimal? TargetValue { get; set; }
    public decimal? ThresholdGreen { get; set; }
    public decimal? ThresholdYellow { get; set; }
    public decimal? ThresholdRed { get; set; }
    public bool AlertEnabled { get; set; }
    public string? AlertRecipientsJson { get; set; }
    public bool IsActive { get; set; } = true;
    public int DisplayOrder { get; set; }
    public long? DashboardId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public long CreatedByUserId { get; set; }
    public DateTimeOffset? ModifiedAt { get; set; }
    public long? ModifiedByUserId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public long? DeletedByUserId { get; set; }
    public Guid VersionToken { get; set; }
    public DateTimeOffset? LastSyncedAt { get; set; }
    public string SyncStatus { get; set; } = string.Empty;
}
